dsgrid/registry/registry_manager.py

```diff
@@ -221,6 +221,86 @@ def load(
         )
         return cls(params)
 
+    def data_sync(self, project_id, dataset_id):
+        """Sync data from the remote dsgrid registry.
+
+        Parameters
+        ----------
+        project_id : str
+            Sync by project_id filter
+        dataset_id : str
+            Sync by dataset_id filter
+        """
+        # TODO: docstring
+        fs_interface = self._params.fs_interface
+        cloud_interface = self._params.cloud_interface
+        offline_mode = self._params.offline
+        dry_run_mode = self._params.dry_run
+        no_prompts = True  # TODO: how do we pass in no_prompts from registry cli parent?
+
+        if not offline_mode:
+            if project_id:
+                if self.project_manager.has_id(project_id):
+                    if dataset_id:
+                        # TODO: self.dataset_manager.has_id() gives us all datasets in the registry. We want all datasets associated with a project. This is an error at the moment.
+                        if self.dataset_manager.has_id(dataset_id):
+                            version = self.dataset_manager.get_current_version(dataset_id)
```

Comment:
This function has a bit of complexity (this is 5 levels of indentation). It could be simplified by separating concepts into different functions.
- One function could find the dataset_id/version pairs for all datasets in a project. Note the the existing method `ProjectConfig.list_registered_dataset_ids`. A similar method could get the versions along with it. (I think you might have a bug here because a project may not be using the latest version of a dataset. The dataset version needs to come from the project config.).
- One function could perform the real work. Consider this signature:
```
def sync_dataset(self, dataset_id, version):
```

Reply:
@daniel-thom I added most of the changes we discussed yesterday. The only one I didn't implement is the one where you mentioned that the project_id and dataset_id components of this function could be broken out into different functions. I don't fully understand how to do this and also not totally sure its necessary given that we have made those other changes. Thoughts?

Reply:
I'm OK with it.